components/src/status_im/ui/components/react.cljs

```diff
@@ -144,18 +144,33 @@
             [text-class (dissoc options-with-style :nested?)]
             nested-text-elements)))
 
+(def text-input-refs (atom #{}))
+
 (defn text-input
   [options text]
-  [(text-input-class)
-   (merge
-    {:underline-color-android  :transparent
-     :max-font-size-multiplier max-font-size-multiplier
-     :placeholder-text-color   colors/text-gray
-     :placeholder              (i18n/label :t/type-a-message)
-     :value                    text}
-    (-> options
-        (update :style typography/get-style)
-        (update :style dissoc :line-height)))])
+  (let [input-ref (atom nil)]
+    (reagent/create-class
+      {:component-will-unmount #(when @input-ref
+                                  (swap! text-input-refs disj @input-ref))
+       :reagent-render
+       (fn [options text]
+         [(text-input-class)
+          (merge
+            {:underline-color-android  :transparent
+             :max-font-size-multiplier max-font-size-multiplier
+             :placeholder-text-color   colors/text-gray
+             :placeholder              (i18n/label :t/type-a-message)
+             :ref                      (fn [r] 
+                                         (when r
+                                           (swap! text-input-refs conj r)
+                                           (reset! input-ref r)
+                                           (when (:ref options) 
```

Comment:
maybe better to call ref anyway even if it's nil?

Reply:
Why would we want that?

Reply:
we don't know, maybe someone handle the nil , and logic will be broken. we just have to keep ref function behaviour

Reply:
Right, i see now, fixed.